### PR TITLE
HDDS-6664. Implements getUri in TrashOzoneFileSystem

### PR DESCRIPTION
**Pull request by pratapchandu**
## What changes were proposed in this pull request?

Fix the following exception being thrown during FileSystem.close()

Exception in thread "Trash Emptier" java.lang.UnsupportedOperationException: fs.getUri() not implemented in TrashOzoneFileSystem
        

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-6664

## How was this patch tested?

Ran relevant checks scripts locally and build-branch was successful with  this change. https://github.com/pratapchandu/ozone/actions/runs/3172304615

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java [hunk 1]
```diff
@@ -123,8 +126,15 @@ private void submitRequest(OzoneManagerProtocolProtos.OMRequest omRequest)
 
   @Override
   public URI getUri() {
-    throw new UnsupportedOperationException(
-        "fs.getUri() not implemented in TrashOzoneFileSystem");
+    URI uri = null;
+    try {
+      uri = new URIBuilder().setScheme(OZONE_URI_SCHEME)
+              .setHost("ofs.trash")
```
Comment (sadanand48) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java:132]: Let's not set a hostname here & set uri to a new URI object , something like o3trash, as we are not using this anywhere and it is just to handle the exception.
```java
public static final String OZONE_O3TRASH_URI_SCHEME = "o3trash";
```

**Comment by Galsza** [comment]
Please fill out the last part of the pull request template on how this change was tested. (And also write unit tests for this, if possible)

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java (hunk 1 above)
Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java]: `return Uri.create(OZONE_O3TRASH_URI_SCHEME + "://")`

Comment (pratapchandu) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java]: URI.create() throws exception 
   Expected authority at index 10: o3trash://
   at java.base/java.net.URI.create(URI.java:883)

Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java]: can we try with return Uri.create(OZONE_O3TRASH_URI_SCHEME + ":///")

Comment (pratapchandu) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/TrashOzoneFileSystem.java]: This works , updated and pushed new commit

**Comment by pratapchandu** [comment]
> Please fill out the last part of the pull request template on how this change was tested. (And also write unit tests for this, if possible)

updated the section

**Review by nandakumar131 — APPROVED** [review]

**Comment by sadanand48** [comment]
Thanks @pratapchandu for the contribution and @nandakumar131 for the review. Merged the change.

**Pull request merged**